docs/rules/rules.tex

```diff
@@ -0,0 +1,1526 @@
+  of transactions\improvement{Valid sequence of transaction hasn't
+    properly been defined, but it's implicit in the definition above,
+    it would maybe be worth making more precise.}:
+  \begin{itemize}
+  \item $\currentdistribution{\emptylist} = \initialdistribution$
+  \item Inductively, and writing
+    \begin{itemize}
+    \item $⌊x⌋$ for the integral part (aka floor) of $x$; \emph{i.e.}
+      $⌊x⌋$ is the unique integer such that $⌊x⌋ ⩽ x < ⌊x⌋+1$
+    \item $\kronecker{⋅}{⋅}$ for
+      \href{https://en.wikipedia.org/wiki/Kronecker_delta}{Kronecker's
+        delta} function:
+      $$
+      \kronecker{i}{j} = \left\{
+        \begin{array}{ll}
+          1 & \mbox{if $i=j$}\\
+          0 & \mbox{otherwise}
+        \end{array}
+      \right.
+      $$
+      In particular, $\kronecker{i}{j}q = q$ if $i=j$ and $0$
+      otherwise. Kronecker's delta is used to select summands: it is
+      the arithmetician's \verb+ifThenElse+ combinator.
+    \item $\stakeof{o}{a} = \repartition{o}(a)\cdot\amount{o} ∈ \mathbb{Q}$
+    \item For inputs of $τ_{n+1}$, we extend $\stakeof{⋅}{⋅}$ to
+      inputs using the validity hypothesis to define $\repartition{i}$ and $\amount{i}$
+    \end{itemize}
+    We define
+\newcommand{\stakerepartitionformula}[4]{∑_{#2∈#3} \left(
+            \begin{aligned}
+              &⌊\stakeof{#2}{#1}⌋\\
+              &\qquad+\\
+              &\raisebox{0pt}[0pt][3.5ex]{$\displaystyle{\kronecker{#1}{\selectrichman{\amount{#2}}}\left(\amount{#2} - ∑_{#4∈\agentids}
+            ⌊\stakeof{#2}{#4}⌋\right)}$}
+            \end{aligned}\right)}
+    \begin{align*}
+      \makebox[1em][l]{$\currentdistribution{τ₁,…,τ_n, τ_{n+1}}(a) =$}\\
+      & \phantom{=} \currentdistribution{τ₁,…τ_n}(a)\\
+      & - \stakerepartitionformula{a}{i}{\inputs{τ_{n+1}}}{b}\\
+      & + \stakerepartitionformula{a}{o}{\outputs{τ_{n+1}}}{b}
+    \end{align*}
+
+    That is, the contribution of unspent output $o$ to agent $a$ is
+    the proportion of the value of $o$ specified by the stake
+    repartition of $o$ rounded \emph{down} to the nearest integer. The
+    remaining unallocated stake goes to the bootstrap stakeholder
+    specified by the $\selectrichman{⋅}$ function, applied to the
+    total value of $o$.
+
+    \begin{treacherous}
+      The treatment of remainder as specified here is only valid for
+      the bootstrap era. Outside of the bootstrap era, the remainder
+      may be different, hence we may have to calculate remainder
+      differently for transactions of the bootstrap eras, and
+      transaction posterior to the bootstrap eras.
+    \end{treacherous}
+  \end{itemize}
+
+  We can then extend $\currentdistribution{⋅}$ on valid chains by defining
+  $$
+  \currentdistribution{b₁,…,b_n} = \currentdistribution{\transactions{b₁},…,\transactions{b_n}}
+  $$
+  Finally we define
+  $$
+  \epochdistribution{b₁,…,b_n} = \currentdistribution{b₁,…,b_{\epochindex{b_1,…,b_n}}}
+  $$
```

Comment:
Isn't there a subtlety here that the stake distribution that determines the leader schedule is not taken at the epoch index, but at `epoch index - k`, because we have to have guaranteed consensus when determining the schedule?

Reply:
As far as I'm aware, stake distributions are cut at each epoch boundaries (both in papers and in the code). If there is a fork across an epoch boundaries, then there are two different schedules. I don't see a problem here.